endpoint/endpoint.go

```diff
@@ -256,7 +256,13 @@ func NewEndpoint(dnsName, recordType string, targets ...string) *Endpoint {
 func NewEndpointWithTTL(dnsName, recordType string, ttl TTL, targets ...string) *Endpoint {
 	cleanTargets := make([]string, len(targets))
 	for idx, target := range targets {
-		cleanTargets[idx] = strings.TrimSuffix(target, ".")
+		// Only trim trailing dots for domain name record types, not for TXT or NAPTR records
+		// TXT records can contain arbitrary text including multiple dots
```

Comment:
```suggestion
		// TXT records can contain arbitrary text including multiple dots
		switch recordType {
	case RecordTypeTXT, RecordTypeNAPTR:
		cleanTargets[idx] = target
	default:
		cleanTargets[idx] = strings.TrimSuffix(target, ".")
	}
```

Reply:
Or move check outside of the targets loop, something like `isTrimTrailingDot` or similar